ext/RMagick/rmimage.c

```diff
@@ -9787,29 +10738,51 @@ Image_pixel_color(int argc, VALUE *argv, VALUE self)
         return Pixel_from_PixelColor(&image->background_color);
     }
 
+    exception = AcquireExceptionInfo();
+
     if (image->storage_class == PseudoClass)
     {
+#if defined(IMAGEMAGICK_7)
+        okay = SetImageStorageClass(image, DirectClass, exception);
+        CHECK_EXCEPTION()
+#else
         okay = SetImageStorageClass(image, DirectClass);
-        rm_check_image_exception(image, RetainOnError);
+        if (rm_should_raise_exception(&image->exception, RetainExceptionRetention))
+        {
+            (void) DestroyExceptionInfo(exception);
+            rm_check_image_exception(image, RetainOnError);
+        }
```

Comment:
This is unrelated to the IM7 upgrade, right? Maybe it would be worth pulling it out into a new PR.

Reply:
This is related to the ImageMagick 7 upgrade because `exception = AcquireExceptionInfo();` has been moved up for IM7.

Reply:
Just pushed another path to make this more readable.